File: crud_spring/crud_spring/src/main/java/com/coding/crud_spring/entity/Mission.java
package com.coding.crud_spring.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


@Entity
@AllArgsConstructor
@NoArgsConstructor
@Data
@Table(name = "mission")
public class Mission {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "endDate", nullable = false)
    private String endDate;

    @Column(name = "freeDays", nullable = false)
    private int freeDays;

    @Column(name = "isForMe", nullable = false)
    private boolean isForMe;

    @Column(name = "sellDays", nullable = false)
    private boolean sellDays;
    @Column(name = "shareMission", nullable = false)
    private boolean shareMission;
    @Column(name = "startDate", nullable = false)
    private String startDate;
    @Column(name = "tjm", nullable = false)
    private int tjm;

    public boolean getIsForMe() {
        return isForMe;
    }

    public void setIsForMe(boolean isForMe) {
        this.isForMe = isForMe;
    }

}

File: src/app/missions/models/mission.ts
// mission.model.ts

export interface Mission {
  id: number;
  endDate: string;
  freeDays: number;
  isForMe: boolean;
  sellDays: number;
  shareMission: boolean;
  startDate: string;
  tjm: number;
  total_days: number;
  idUser: number;
  name: string;
  descriptif: string;
  client: {
    id: number;
    nameClient: string;
    mail: string;
    address: string;
    contactName: string;
    tel: string;
    commentaires: string;
  };
  active: boolean;
  totalDays: number;
}
